libarchive/archive_read_support_format_zip.c

```diff
@@ -1144,7 +1144,8 @@ zip_read_local_file_header(struct archive_read *a, struct archive_entry *entry,
 			    (intmax_t)zip_entry->compressed_size);
 			ret = ARCHIVE_WARN;
 		}
-		if (zip_entry->uncompressed_size == 0) {
+		if (zip_entry->uncompressed_size == 0 ||
+			zip_entry->uncompressed_size == 0xffffffff) {
```

Comment:
Do you have any special reason why writing out the hex and not explicitly using "== -1" ?

Reply:
I was following convention in the rest of the code that uses hex values. Also, the field is unsigned so you would have to cast to unsigned

Reply:
thx! I'm not sure what archiver was using -1 instead of 0, but I've seen several examples